### DISCUSSION TITLE
Mocking Component.render_to_response

### DISCUSSION THREAD

**Discussion by TheSteveBurgess** [django-components/django-components#844] [Q&A] [answered]
This might be my lack of Python knowledge here, but I'm struggling to mock the render_to_response method on Component.

I want to be able to check that the correct context values are being passed to the method (hence wanting to mock it).

However,  it looks like it's doing some juggling to allow the method to both be a class method and an instance method depending on how its called

```
self.render_to_response = types.MethodType(self.__class__.render_to_response.__func__, self)  # type: ignore
```

Am i missing something obvious? Is there a specific way you would go about mocking something like this in Python?

**Comment by JuroOravec** [comment, marked as answer]
So far I always tested only the output HTML, not the context, so I haven't tried mocking `render_to_response`.

Some options that come to mind:
1. Mock `self._render()` instead - this is the normalized entrypoint, and it doesn't get re-applied like the public `render` and `render_to_response` do
2. Check for the given context in `get_context_data()`, e.g.
    ```py
    # Setup
    captured = None
    class MyTable(Component):
        def get_context_data(self):
            captured = self.input.context

    # Act
    MyTable.render_to_response({ "data": [1,2,3] })

    # Assert
    assert isinstance(captured, Context)
    ```

**Reply by TheSteveBurgess** [reply]
Ahh self._render() looks like it should do the trick :)

Thanks!
